scripts/create_changelog.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+This script is used in 'Generate Changelog' workflow, defined in the 'generate_changelog.yml' file
+to automate the process of creating changelogs after each stable release.
+
+It fetches release data, extracts relevant pull request (PR) information, and generates a detailed changelog
+and add its to src/changelogs.
+The script uses the GitHub REST API to collect information about contributors, PR authors, and reviewers.
+
+Additional Note:
+- The script requires a GitHub personal access token (PAT) stored in an environment variable named `GITHUB_PAT`.
+"""
+
+import requests
+import re
+from dotenv import load_dotenv
+import os
+import argparse
+import xml.etree.ElementTree as ET
+from unidecode import unidecode
+
+load_dotenv()
+GITHUB_PAT = os.getenv('GITHUB_PAT')
+BASE_URL = r"https://api.github.com/repos/sagemath/sage"
+HEADERS = {'Authorization': f'token {GITHUB_PAT}', }
+AUTOMATED_BOTS = ['dependabot[bot]', 'github-actions', 'renovate[bot]']
+
+# Maps the github username to contributer name
+git_to_name = {}
+
+# Stores unique names of contributors
+all_contribs = set([])
+
+# Stores unique names of contributors
+first_contribs = set([])
+
+# Stores information for all the prs across all pre-releases
+# Maps tag of pre-release to its info
+all_info = {}
+
+
+def map_git_to_names():
+    tree = ET.parse('conf/contributors.xml')
+    root = tree.getroot()
+    for c in root.findall('contributor'):
+        name = c.get('name')
+        git = c.get('github')
+        if git and name:
+            git_to_name[git] = unidecode(name)
+
+
+def update_names():
+    """
+    Replace the github usernames with real names. If name is not found in contributors.xml,
+    then github usernames are used in the form @<github-username>
+    """
+    for tag in all_info:
+        for pr in all_info[tag]:
+            pr['creator'] = git_to_name.get(pr['creator'], f"@{pr['creator']}")
+            pr['authors'] = [git_to_name.get(a, f"@{a}") for a in pr['authors']]
+            pr['reviewers'] = [git_to_name.get(r, f"@{r}") for r in pr['reviewers']]
+    global all_contribs
+    global first_contribs
+    all_contribs = set([git_to_name.get(c, f"@{c}") for c in all_contribs])
+    first_contribs = set([git_to_name.get(c, f"@{c}") for c in first_contribs])
+
```

Comment:
It seems that we can retrieve the github user's real name via github api from his/her github account name. If this is the case, then we may create a new entry in `conf/contributors.xml` and then use the real name instead of the github account name in the created changelog. For example, we may create this entry
```
<contributor
 name="Marie Bonboire"
 github="marizee"/>
```
and use "Marie Bonboire" instead of `@marizee`. The entries in `contributors.xml` are listed in alphabetical order in the last name. We may use the fact to find the right place to put the new entry in.

But this is out of the scope of this PR. You may decide to do this extra work or not. Just let me know.

Reply:
Yes, this is doable, but some users might not have their real name on their Github Profile, in such case we seem to have no other option than to user their github username in the format `@github-username`.
I will work on it